test/quantization/test_quantized_tensor.py

```diff
@@ -257,6 +307,41 @@ def quantize_ref(data, scales, zero_points):
         self.assertTrue(np.allclose(qr.int_repr(), ref))
         self.assertTrue(np.allclose(r.numpy(), dequant_tensor.numpy(), atol=1))
 
+        def _quantize_per_channel_ref_nd(data, scales, zero_points):
+            dims = data.size()
+            data = data.view(-1, dims[1], np.prod(dims[2:]))
+            res = torch.empty_like(data)
+            quant_min, quant_max = 0, 255
+            for i in range(res.size()[0]):
+                for j in range(res.size()[1]):
+                    for k in range(res.size()[2]):
+                        inv_scale = 1.0 / scales[j]
+                        res[i][j][k] = \
+                            np.clip(np.round(data[i][j][k] * inv_scale + zero_points[j]), quant_min, quant_max)
+            res = res.view(*dims)
+            return res
+
+        # Check 4D tensor with 2 different memory formats.
+        r = torch.rand(3, 2, 4, 5, dtype=torch.float) * 4
+
+        for memory_format in [torch.contiguous_format, torch.channels_last]:
+            ref_res = _quantize_per_channel_ref_nd(r, scales, zero_points)
+            r = r.contiguous(memory_format=memory_format)
+            qr = torch.quantize_per_channel(r, scales, zero_points, axis, torch.quint8)
+            rqr = qr.dequantize()
+            self.assertTrue(np.allclose(qr.int_repr(), ref_res))
+            self.assertTrue(np.allclose(r.numpy(), rqr.numpy(), atol=2 / np.min(scales.numpy())))
+
+        # Check 5D tensor.
+        r = torch.rand(3, 2, 4, 5, 7, dtype=torch.float) * 4 - 2
+
+        for memory_format in [torch.contiguous_format, torch.channels_last_3d]:
+            ref_res = _quantize_per_channel_ref_nd(r, scales, zero_points)
+            r = r.contiguous(memory_format=memory_format)
+            qr = torch.quantize_per_channel(r, scales, zero_points, axis, torch.quint8)
+            rqr = qr.dequantize()
+            self.assertTrue(np.allclose(qr.int_repr(), ref_res))
+            self.assertTrue(np.allclose(r.numpy(), rqr.numpy(), atol=2 / np.min(scales.numpy())))
```

Comment:
checks looks pretty much the same as well, can we merge them into a local function?